src/GraphQL/Types/Composite/AutoRegisteringObjectGraphType.cs

```diff
@@ -169,10 +230,12 @@ protected virtual IEnumerable<MemberInfo> GetRegisteredMembers()
                 _excludedProperties);
             var methods = typeof(TSourceType).GetMethods(BindingFlags.Public | BindingFlags.Instance | BindingFlags.DeclaredOnly)
                 .Where(x =>
-                    !x.ContainsGenericParameters && // exclude methods with open generics
-                    !x.IsSpecialName &&             // exclude methods generated for properties
-                    x.ReturnType != typeof(void) && // exclude methods which do not return a value
-                    x.ReturnType != typeof(Task));  // exclude methods which do not return a value
+                    !x.ContainsGenericParameters &&               // exclude methods with open generics
+                    !x.IsSpecialName &&                           // exclude methods generated for properties
+                    x.ReturnType != typeof(void) &&               // exclude methods which do not return a value
+                    x.ReturnType != typeof(Task) &&               // exclude methods which do not return a value
+                    x.GetBaseDefinition() == x &&                 // exclude methods which override an inherited class' method (e.g. GetHashCode)
+                    x.Name != "Equals" && x.Name != "<Clone>$");  // exclude methods generated for record types
```

Comment:
Bug fix: added two lines here to support record types properly.  I could not find a better way to identify the auto-generated methods. Tests added for both input and output graph types.

Reply:
`var isRecord = x.GetMethods().Any(m => m.Name == "<Clone>$");`

`
x.GetBaseDefinition() == x &&                 // exclude methods which override an inherited class' method (e.g. GetHashCode)
x.Name != "Equals" && x.Name != "<Clone>$");
`
->
`
x.GetMethods().Any(m => m.Name == "<Clone>$") &&
x.Name != "Equals" && x.Name != "<Clone>$");
`

Actually `Equals` may be a valid method to map into field so it should be excluded only for record types.

Reply:
ok

Reply:
how about excluding methods with this signature:

```cs
public bool Equals(TSourceType other)
```

where TSourceType is the type of the clr type being examined?

Reply:
I implemented as I suggested, excluding by convention/signature rather than more specifically by whether it's detected as a record type.  please review and lmk if this is okay.

Reply:
Ok; updated. Please review again.

Reply:
I mean that you should not take the responsibility for all "standard" methods. Why only `Equals`? If you try to eliminate `Equals` then you should to deal with `GetHashCode`, `ToString`.

Reply:
Record is a special case - we 100% know that records have special methods that are compiler-generated. So we 100% sure that such methods should not be considered for mapping.

Reply:
> Record is a special case - we 100% know that records have special methods that are compiler-generated. So we 100% sure that such methods should not be considered for mapping.

Are you requesting further changes?

Reply:
If anyone wants a field called GetHashCode or ToString, they can probably use `new`, and even if not certainly they can create a method with any other name and simply use the `[Name]` attribute to set the graphql name to "GetHashCode" or "ToString".  I would personally take this approach with Equals where it matches the well-known signature.  But okay if you would rather include by default.

Reply:
> the virtual that overridden methods are always skipped

where?

Reply:
`x.GetBaseDefinition() == x` is false if it's a overridden virtual method.

Reply:
Note that methods from inherited classes are skipped because DeclaredOnly enum is used with GetMethods.  But if it's overridden then it shows up anyway in GetMethods, and that's where `x.BaseDefinition() == x` comes into play.

Reply:
ok